ui/js/models/perf/alerts.js

```diff
@@ -359,6 +360,39 @@ treeherder.factory('PhAlerts', [
                         return newAlertSummaryId;
                     });
                 });
-            }
+            },
+            nudgeAlert: (dataPoint, direction, resultSetData) => {
+
+                function findPushIdNeighbour(pushId, resultSetData, direction) {
+                    let pushIdIndexOf;
+                    let relativePos;
+
+                    // retriggers duplicate pushes
+                    if (direction === 'left') {
+                        pushIdIndexOf = 'indexOf';
+                        relativePos = -1;
+                    } else {
+                        pushIdIndexOf = 'lastIndexOf';
+                        relativePos = 1;
+                    }
+
+                    const pushIdIndex = resultSetData[pushIdIndexOf](pushId);
+                    return resultSetData[pushIdIndex+relativePos];
+                }
```

Comment:
I would put this logic in the upper-end function and then pass the new push id to this function. Just seems a little cleaner this way. It may also be worth getting @camd to look at this code and see if he has any suggestions on how to structure it, given that we're moving away from angular.